src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs

```diff
@@ -342,7 +344,17 @@ private static EndpointMetadataCollection BuildEndpointMetadata(ActionDescriptor
                 {
                     if (actionConstraint is HttpMethodActionConstraint httpMethodActionConstraint)
                     {
-                        metadata.Add(new HttpMethodEndpointConstraint(httpMethodActionConstraint.HttpMethods));
+                        var httpEndpointConstraint = new HttpMethodEndpointConstraint(httpMethodActionConstraint.HttpMethods);
+
+                        // yolo
+                        if (httpMethodActionConstraint.GetType().Name == "CorsHttpMethodActionConstraint")
```

Comment:
Sorry not sorry.

Reply:
But seriously CORS is off in its own project so maybe we need to think about extensiblity model for EndpointDataSource and handling situations like this.